### ISSUE TITLE
Could be possible to host on serverless/heroku?

### ISSUE THREAD

**Issue by tiero** [globaleaks/Tor2web#371]
Hi I need a tor proxy, but not always on. I was researching how to make ti work in serveless fashion (eg. Lambda + Api Gateway) but seems impossibile. 

Do you think could be possible to host it on Heorku like platform? Thanks

**Comment by evilaliv3** [comment]
@tiero: As far that it can run Debian Buster (10) yes.

Which is the use case if i may ask?

**Comment by tiero** [comment]
TL;DR I want to parasitize free tiers of cloud providers, without paying for a machine always on.

Long story:

I have development machines at "home", where I don't have a public ip address, to overcome this an Onion service solved it. The problem is that in is some context I don't have the possibility to run a tor client (eg. CI pipelines) so a simple tor proxy should work. 

I don't want to use public tor2web websites if I can and also most of them don't support yet v3 onion services. 

I saw that building from master should be possible, so I was looking into Heroku, and I was asking if other already did it before.

**Comment by evilaliv3** [comment]
I see, i do not think so but please let us know if your experiment proceed well and we would be happy to publish scripts for heroku officially.

**Comment by fpietrosanti** [comment]
The problem is that, afaik, you can't run native code on Heroku, so you could run Python but you wont be able to execute Tor, that's a dependency of Tor2web as a third party independent native code software executable.

**Comment by tiero** [comment]
I saw that using buuldpacks something can be worked out eg. https://elements.heroku.com/buildpacks/todokku/heroku-buildpack-tor-proxy 

I'll try to experiment and keep you posted

**Comment by fpietrosanti** [comment]
> I saw that using buuldpacks something can be worked out eg. https://elements.heroku.com/buildpacks/todokku/heroku-buildpack-tor-proxy
> 
> I'll try to experiment and keep you posted

If you make it working, it means that we will be able also to run GlobaLeaks on Heroku https://github.com/globaleaks/globaleaks and as you can make it i'll offer a huge dinner! :>
